core/unit_test/TestHostSharedPtrAccessOnDevice.hpp

```diff
@@ -52,14 +52,15 @@ using Kokkos::Impl::HostSharedPtr;
 namespace {
 
 class Data {
-  Kokkos::Array<char, 64> d;
+  char d[64];
 
  public:
-  KOKKOS_FUNCTION void write(char const* c) {
-    for (int i = 0; i < 64 && c; ++i, ++c) {
-      d[i] = *c;
-    }
+  // Because strncpy is not supported within device code
+  KOKKOS_FUNCTION void my_strncpy(char* dst, const char* src, size_t cnt) {
```

Comment:
This isn't the behavior of `strncpy`, so we should change the name.  `strncpy` always writes `cnt` number of bytes into `dst`.  However, if a `\0` is detected in `src`, the rest of `src` is no longer read (as it may not even exist), and `\0` is written to the rest of the bytes in `dst`.

Alternatively, we could implement the `strncpy` behavior (caution: not tested):

```
while (cnt > 0 && *src != '\0`) {
    *dst++ = *src++;
    --cnt;
}
while (cnt > 0) {
    *dst++ = '\0';
   --cnt;
}
```

or the more clever

```
while (cnt-- > 0 && ((*dst++ = *src++) != '\0') {}
while (cnt-- > 0) *dst++ = '\0';
```

Which fixes a bug if `c` is not a large enough string.

Reply:
It should also now be a `static` member function.

Reply:
... or a free function.

Reply:
Fixed